docs/app/pages/components/components-sections/input-controls/typeahead/snippets/app.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+import 'chance';
+import { TypeaheadOptionEvent, TypeaheadKeyService } from '@ux-aspects/ux-aspects';
+
+@Component({
+    selector: 'app',
+    templateUrl: 'app.component.html',
+    styleUrls: ['./app.component.css'],
+})
+export class AppComponent {
+
+    values: ReadonlyArray<string> = [];
+
+    dropdownOpen: boolean = false;
+    selectOnEnter: boolean = false;
+    dropDirection: 'up' | 'down' = 'down';
+    selectFirst: boolean = false;
+
+    value: string = '';
+    input: string = '';
+
+    loadOptionsFn = this.loadOptions.bind(this);
+
+    /** Load the options and filter the them */
+    loadOptions(pageNum: number, pageSize: number, filter: string): ReadonlyArray<string> {
+        return this.values.filter(tag => tag.toLowerCase().indexOf(filter.toLowerCase()) !== -1).slice(pageNum * pageSize, (pageNum + 1) * pageSize);
```

Comment:
This line is causing scrollbars to appear on the sample code snippet. Split this over multiple lines, eg:

```typescript
return this.values.filter(tag => tag.toLowerCase().indexOf(filter.toLowerCase()) !== -1)
    .slice(pageNum * pageSize, (pageNum + 1) * pageSize);
```